pkg/serializer/internal/metrics/service_checks_test.go

```diff
@@ -119,10 +119,8 @@ func TestPayloadsEmptyServiceCheck(t *testing.T) {
 }
 
 func TestPayloadsServiceChecks(t *testing.T) {
-
-	maxPayloadSize := config.Datadog.GetInt("serializer_max_payload_size")
-	config.Datadog.SetDefault("serializer_max_payload_size", 200)
-	defer config.Datadog.SetDefault("serializer_max_payload_size", maxPayloadSize)
+	config.Datadog.SetForSource("serializer_max_payload_size", 200, config.SourceSelf)
+	defer config.Datadog.UnsetForSource("serializer_max_payload_size", config.SourceSelf)
```

Comment:
I may be missing something, but this isn't equivalent to the old code, if any init code also uses `SourceSelf` it will be erased. Would it make sense to have `SourceTest` too?

Reply:
This does indeed mean that if init code uses SourceSelf to set "serializer_max_payload_size" then it would be erased by this test
It might be ok as this is a unit test, but the idea of having a SourceTest deserves to be thought upon

For now the tests pass with those changes, the agent shared components team has a ticket to check each call to the .Set() method to ensure the proper Source is used